src/core/world_state.h

```diff
@@ -0,0 +1,68 @@
+#ifndef CORE_WORLD_STATE_H
+#define CORE_WORLD_STATE_H
+
+#include "utils/common_utils.h"
+
+typedef enum{
+	ELEMENT_ACTIVE,
+	ELEMENT_IN_ROBOT,
+	ELEMENT_DONE,
+	ELEMENT_LOST
+}element_status_t;
+
+typedef enum{
+	MECHANISM_FULL,
+	MECHANISM_ACTIVE,
+	MECHANISM_ERROR
+}mechanism_status_t;
+
+typedef enum{
+	ELEMENT_COMMON,
+	ELEMENT_ALLY
+}element_distribution_t;
+
+typedef struct{
+	mechanism_status_t status;
+	int elements_count;
+}mechanism_t;
+
+typedef struct{
+	element_status_t status;
+	element_distribution_t distribution;
+	char name[32];
+	point_t position;
+}element_t;
+
+
+/// @todo todo remove n from ifndef :D
```

Comment:
Nothing against this comment, or the change it is talking about, but does it really matter?